### DISCUSSION TITLE
Unable to get a new action working

### DISCUSSION THREAD

**Discussion by mslourens** [Budibase/budibase#2113] [Help] [answered]
I am trying to implement #2102. 

I added a new button action definition to the EventsEditor actions, defined a new ActionType constant, created an action handler in buttonActions in the client, but when I want to act on the action in the InnerForm.svelte, the action is not retrieved. Somehow the action in the executeActionHandler has no componentId. 
How does the componentId gets passed to the action parameters? And is this parameter required to get the action working? 

You can find my changes in [this branch](https://github.com/mslourens/budibase/tree/clear_form_action)

**Comment by aptkingston** [comment, marked as answer]
Had a quick look at your branch there. You'll need to make a component to accept parameters for the new action. The `ValidateForm` component is a perfect candidate to copy, because it also just accepts a single parameter which is the form to validate. So if you copy that to `ClearForm` and update the `component` key in the definitions in `actions/index.js` to use your new component, then you'll have a component ID parameter to work with when inside `buttonActions.js` in the client library.

**Reply by mslourens** [reply]
I used the `ValidateForm` as an example, but I thought I didn't need the form selector. In order to get it working, I also had to adjust the manifest file from the standard-components, otherwise the app won't show the form in the form selector.
